test/generation.test.ts

```diff
@@ -26,72 +26,40 @@ describe('Generation tests', () => {
       sc.forall('a', fc.integer(-10, 10)).forall('b', fc.integer(-10, 10)).prng.generator).to.be.true
   })
 
-  it('Generator propagates to arbitraries without generator specification', () => {
-    const sc = fc.scenario()
-
-    const arb1 = fc.integer(-10, 10)
-    sc.forall('a', arb1)
-    expect(sc.prng.generator === arb1.generator).to.be.true
-
-    const arb2 = fc.integer(-10, 10)
-    sc.forall('a', arb1).forall('b', arb2)
-    expect(sc.prng.generator === arb1.generator).to.be.true
-    expect(sc.prng.generator === arb2.generator).to.be.true
-  })
-
   it('Generator generates different values for two similar arbitraries without generator specification', () => {
-    const arb1 = fc.integer(-10, 10)
-    const arb2 = fc.integer(-10, 10)
-    fc.scenario().forall('a', arb1).forall('b', arb2)
+    const sc1 = fc.scenario().forall('a', fc.integer(-10, 10))
+    const sc2 = sc1.forall('b', fc.integer(-10, 10))
 
-    expect(arb1.sampleUniqueWithBias() !== arb2.sampleUniqueWithBias()).to.be.true
+    expect(sc1.getCache() !== sc2.getCache()).to.be.true
```

Comment:
What tyoe of object does `getCache()` return? Are you sure `!==` is what you want (i.e. to compare references and not content?)